src/main/java/org/sagebionetworks/web/client/widget/entity/download/Uploader.java

```diff
@@ -791,25 +791,30 @@ public void onSuccess(Entity result) {
         entity = result;
         view.showInfo(DisplayConstants.TEXT_LINK_SUCCESS);
         if (successHandler != null) {
-          synapseClient.getEntityBenefactorAcl(
+          jsClient.getEntityBenefactorAcl(
             result.getId(),
             new AsyncCallback<AccessControlList>() {
               @Override
               public void onSuccess(AccessControlList benefactorAcl) {
-                successHandler.onSuccessfulUpload(benefactorAcl.getId());
+                if (benefactorAcl.getId().equals(entity.getId())) {
+                  // Don't show the ACL modal if the entity is its own benefactor
+                  successHandler.onSuccessfulUpload(null);
+                } else {
+                  successHandler.onSuccessfulUpload(benefactorAcl.getId());
+                }
+                entityUpdated();
```

Comment:
Move `entityUpdated` call inside callback to fix race condition issue where the Uploader would be partially torn down before `successHandler` is called